ethereum/eip712/eip712_test.go

```diff
@@ -0,0 +1,309 @@
+			fee: txtypes.Fee{
+				Amount:   makeCoins(t, "aphoton", math.NewInt(2000)),
+				GasLimit: 20000,
+			},
+			memo: "",
+			msgs: []sdk.Msg{
+				govtypes.NewMsgVote(
+					createTestAddress(t),
+					5,
+					govtypes.OptionNo,
+				),
+			},
+			accountNumber: 25,
+			sequence:      78,
+			expectSuccess: true,
+		},
+		{
+			title: "Standard MsgDelegate",
+			fee: txtypes.Fee{
+				Amount:   makeCoins(t, "aphoton", math.NewInt(2000)),
+				GasLimit: 20000,
+			},
+			memo: "",
+			msgs: []sdk.Msg{
+				stakingtypes.NewMsgDelegate(
+					createTestAddress(t),
+					sdk.ValAddress(createTestAddress(t)),
+					makeCoins(t, "photon", math.NewInt(1))[0],
+				),
+			},
+			accountNumber: 25,
+			sequence:      78,
+			expectSuccess: true,
+		},
+		{
+			title: "Standard MsgWithdrawDelegationReward",
+			fee: txtypes.Fee{
+				Amount:   makeCoins(t, "aphoton", math.NewInt(2000)),
+				GasLimit: 20000,
+			},
+			memo: "",
+			msgs: []sdk.Msg{
+				distributiontypes.NewMsgWithdrawDelegatorReward(
+					createTestAddress(t),
+					sdk.ValAddress(createTestAddress(t)),
+				),
+			},
+			accountNumber: 25,
+			sequence:      78,
+			expectSuccess: true,
+		},
+		{
+			title: "Two MsgVotes",
+			fee: txtypes.Fee{
+				Amount:   makeCoins(t, "aphoton", math.NewInt(2000)),
+				GasLimit: 20000,
+			},
+			memo: "",
+			msgs: []sdk.Msg{
+				govtypes.NewMsgVote(
+					createTestAddress(t),
+					5,
+					govtypes.OptionNo,
+				),
+				govtypes.NewMsgVote(
+					createTestAddress(t),
+					25,
+					govtypes.OptionAbstain,
+				),
+			},
+			accountNumber: 25,
+			sequence:      78,
+			expectSuccess: false, // Multiple messages (check for multiple signers in AnteHandler)
+		},
+		{
+			title: "MsgSend + MsgVote",
+			fee: txtypes.Fee{
+				Amount:   makeCoins(t, "aphoton", math.NewInt(2000)),
+				GasLimit: 20000,
+			},
+			memo: "",
+			msgs: []sdk.Msg{
+				govtypes.NewMsgVote(
+					createTestAddress(t),
+					5,
+					govtypes.OptionNo,
+				),
+				banktypes.NewMsgSend(
+					createTestAddress(t),
+					createTestAddress(t),
+					makeCoins(t, "photon", math.NewInt(50)),
+				),
+			},
+			accountNumber: 25,
+			sequence:      78,
+			expectSuccess: false, // Multiple messages
+		},
+	}
+
+	for _, tc := range testCases {
+		for _, signMode := range signModes {
+			privKey, pubKey := createTestKeyPair(t)
+
+			// Init tx builder
+			txBuilder := clientCtx.TxConfig.NewTxBuilder()
+
+			// Set gas and fees
+			txBuilder.SetGasLimit(tc.fee.GasLimit)
+			txBuilder.SetFeeAmount(tc.fee.Amount)
+
+			// Set messages
+			err := txBuilder.SetMsgs(tc.msgs...)
+			require.NoError(t, err)
+
+			// Set memo
+			txBuilder.SetMemo(tc.memo)
+
+			// Prepare signature field
+			txSigData := signing.SingleSignatureData{
+				SignMode:  signMode,
+				Signature: nil,
+			}
+			txSig := signing.SignatureV2{
+				PubKey:   pubKey,
+				Data:     &txSigData,
+				Sequence: tc.sequence,
+			}
+
+			err = txBuilder.SetSignatures([]signing.SignatureV2{
```

Comment:
nit, is easier to read if its on one line

```suggestion
			err = txBuilder.SetSignatures([]signing.SignatureV2{	txSig}...)
```